l5kit/l5kit/simulation/unroll.py

```diff
@@ -119,11 +140,12 @@ def __init__(self, sim_cfg: SimulationConfig, dataset: EgoDataset,
         :param model_ego: the model to be used for ego
         :param model_agents: the model to be used for agents
         :param keys_to_exclude: keys to exclude from input/output (e.g. huge blobs)
+        :param verify_model: flag to verify the ego/agent model when use_ego_gt/use_agent_gt is False
```

Comment:
Why is this required?

Reply:
The ClosedLoopSimulator requires both the SimulationDataset and an ego_model when use_ego_gt = False. However, in gym environment, the SimulationDataset (part of environment) is independent of the ego_model (the policy). So, the ClosedLoopSimulator should not have an ego_model (it is not part of environment), and this will raise an error. 

The verify_model flag helps to check whether we want to verify the presence of ego_model (as in L5Kit) or not (for gym). Does this explanation help?

Reply:
A bit. It seems we're introducing a lot of complexity though

Reply:
Without this flag, we might have to end up writing a class very similar to ClosedLoopSimulator, but without the ego and agent models as the model (policy) in L5Gym is decoupled from the simulator (the environment).

I believe the verification flag is a small added complexity that allows to re-use the entire ClosedLoopSimulator for L5 gym environment. What do you think?

Reply:
should we have this more explicit? 
what about:
- defining an IntEnum with 2 modes (gym, l5kit)
- taking that as a param and perform the check based on that

what do you think?

Reply:
Thats certainly a better way. I will do this!